single-host-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/singlehostclient/ServerConfiguration.java

```diff
@@ -69,6 +74,7 @@ public ServerConfiguration(
         this.healthReadTimeoutInMilliseconds = healthReadTimeoutInMilliseconds;
         this.failureResponseCountThreshold = failureResponseCountThreshold;
         this.requestTransformer = requestTransformer;
+        this.protocol = protocol;
```

Comment:
I would validate the value of the protocol either http or https since nothing else is expected and throw an illegalargumentexception if invalid

Reply:
yep, done

Reply:
very minor - `PROTOCOLS` could be names SUPPORTED_PROTOCOLS and the error message could use the list instead of re-enumerating them

Reply:
yep, done